adal/src/telemetry/java/com/microsoft/aad/adal/AggregatedDispatcher.java

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package com.microsoft.aad.adal;
+
+import android.util.Pair;
+
+import java.util.ArrayList;
+import java.util.List;
+
+class AggregatedDispatcher extends DefaultDispatcher {
+
+    AggregatedDispatcher(final IDispatcher dispatcher) {
+        super(dispatcher);
+    }
+
+    @SuppressWarnings("unchecked")
+    synchronized void flush(final String requestId) {
+        final List<Pair<String, String>> dispatchList = new ArrayList<>();
+        if (getDispatcher() == null) {
+            return;
+        }
+
+        final List<IEvents> events = getObjectsToBeDispatched().remove(requestId);
+        if (events == null || events.isEmpty()) {
+            return;
+        }
+
+        dispatchList.addAll(events.get(0).getEvents());
```

Comment:
Just a reminder, don't forget to add comments here. Just briefly explain why we do a subList, it will be easy for people to understand when reading the code.